modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/NewTopic.es.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal, {useModal} from '@clayui/modal';
+import React, {useState} from 'react';
+
+export default () => {
+	const [visible, setVisible] = useState(false);
+
+	return (
+		<>
+			<NewTopicModal
+				onClose={() => setVisible(false)}
+				visible={visible}
+			/>
+			<ClayButton
+				displayType="secondary"
+				onClick={() => setVisible(true)}
+			>
+				<ClayIcon symbol="plus" />
+				{Liferay.Language.get('new-topic')}
+			</ClayButton>
+		</>
+	);
+
+	function NewTopicModal({onClose, visible}) {
```

Comment:
Good try but don't extract the function for such small component or use it as function

Reply:
You didn't change it :P